### PR TITLE
Fix for schema sequences

### PR DESCRIPTION
**Pull request by JPinkney**
Fixes: https://github.com/redhat-developer/yaml-language-server/issues/196

The idea of this PR is described in: https://github.com/redhat-developer/yaml-language-server/pull/81#issuecomment-548788330

cc @andxu @itowlson

### CODE REVIEW

**Comment by itowlson** [comment]
Thanks @JPinkney - it's the weekend here but I'll take a look on Monday.  Thanks heaps for the quick turnaround!

**Comment by itowlson** [comment]
I don't feel very qualified to comment on the code in detail, but it definitely looks like a solid approach and the implementation looks good.  I'll install a local build and see if fixes the downstream problem with the k8s extension.  Thanks!

**Comment by andxu** [comment]
LGTM

**Comment by itowlson** [comment]
I'm not sure if I've got something wrong in the build chain (`yaml-language-server -> vscode-yaml -> vscode-kubernetes-tools`) but this still doesn't seem to be working for me when I build it from source.  I do still get intellisense on single documents but nothing on multi-documents.

**Comment by itowlson** [comment]
I'm not sure how to track down the problem or help give any direction given how indirect the chain is - can't think of anything except sprinkling printfs and that doesn't seem very efficient... @JPinkney let me know if there's anything I can do to help validate whether there's a problem and if so to diagnose it.

File: src/languageservice/services/yamlSchemaService.ts [hunk 1]
```diff
@@ -43,7 +43,12 @@ export class YAMLSchemaService extends JSONSchemaService {
             }
 
             if (schemas.length > 0) {
-                return super.createCombinedSchema(resource, schemas).getResolvedSchema();
+                return super.createCombinedSchema(resource, schemas).getResolvedSchema().then(schema => {
+                    if (schema.schema && schema.schema.schemaSequence && schema.schema.schemaSequence[doc.currentDocIndex]) {
+                        return new ResolvedSchema(schema.schema.schemaSequence[doc.currentDocIndex]);
+                    }
+                    return schema;
+                });
```
Comment (andxu) [inline src/languageservice/services/yamlSchemaService.ts]: add the code handing sequence for custom schema provider at line 64:
```
if (this.customSchemaProvider) {
            return this.customSchemaProvider(resource)
                       .then(schemaUri => {
                           if (!schemaUri) {
                               return resolveSchema();
                           }
                           return this.loadSchema(schemaUri)
                               .then(unsolvedSchema => this.resolveSchemaContent(unsolvedSchema, schemaUri, [])).then(schema => {
                                if (schema.schema && schema.schema.schemaSequence && schema.schema.schemaSequence[doc.currentDocIndex]) {
                                    return new ResolvedSchema(schema.schema.schemaSequence[doc.currentDocIndex]);
                                }
                                return schema;
                               });
                        })
                       .then(schema => schema, err => resolveSchema());
        } else {
            return resolveSchema();
        }
```

**Comment by JPinkney** [comment]
@andxu @itowlson Can you try again now, it wasn't resolving `$ref` properly before

**Comment by itowlson** [comment]
Looking good now!  I'm using @andxu's test pod/deployment YAML and the hints and errors are looking appropriate to me.  Thanks!

**Comment by andxu** [comment]
@itowlson I tested the new code and it works now

**Comment by itowlson** [comment]
@JPinkney Is there a timeframe for getting this into a release please?

**Comment by JPinkney** [comment]
@itowlson I'll try for Friday, I'm a little short on time since I'm preparing for KubeCon

**Comment by itowlson** [comment]
Terrific - thanks @JPinkney!

**Pull request merged**